src/native/libs/System.Security.Cryptography.Native/pal_evp.c

```diff
@@ -10,31 +10,55 @@
 
 #define SUCCESS 1
 
-static const EVP_MD* g_evpFetchMd5 = NULL;
-static pthread_once_t g_evpFetch = PTHREAD_ONCE_INIT;
-
-static void EnsureFetchEvpMdAlgorithms(void)
-{
-    // This is called from a pthread_once - this method should not be called directly.
-
 #ifdef NEED_OPENSSL_3_0
-    if (API_EXISTS(EVP_MD_fetch))
-    {
-        ERR_clear_error();
-
-        // Try to fetch an MD5 implementation that will work regardless if
-        // FIPS is enforced or not.
-        g_evpFetchMd5 = EVP_MD_fetch(NULL, "MD5", "-fips");
+#define SETUP_MD_FETCH(export, fn, name, query) \
```

Comment:
I'd probably call these `DECLARE_`. `DEFINE_`, or `BUILD_` rather than `SETUP_`, since SETUP can be read as "I belong to the setup process", or "set up this function so we can call it".  Clearly it means the latter, but I keep reading it as a noun ("setup") instead of a compound verb ("set up").